cmd/snap-bootstrap/partition/encrypt.go

```diff
@@ -136,3 +162,63 @@ func cryptsetupClose(name string) error {
 	}
 	return nil
 }
+
+func cryptsetupAddKey(key EncryptionKey, rkey RecoveryKey, node string) error {
+	// create a named pipe to pass the recovery key
+	fpath := filepath.Join(dirs.SnapRunDir, "tmp-rkey")
```

Comment:
A silly question maybe, should we use a well known path here, or rather create a pipe, pass that in ExtraFiles and use `luksAddKey <node> -q --key-file - --key-slot 1 /proc/self/fd/3` ?

Reply:
@chrisccoulson may add these to secboot as helpers so I'll leave them as they are for now and we can adjust details there when they're implemented.